src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -0,0 +1,575 @@
+    const auto& scalar_tile_body = std::dynamic_pointer_cast<TileEmitter>(scalar_tile.first)->get_nested_code();
+
+    // remove data ptr regs from the pool, since they should be preserved
+    const auto& data_ptr_reg_idxs(out);
+
+    // It is critical that reg_outer_amount and reg_inner_amount represent the
+    // first two runtime arguments, since they are used to calculating offsets
+    Reg64 reg_outer_amount = Reg64(static_cast<int>(in[3]));
+    Reg64 reg_inner_amount = Reg64(static_cast<int>(in[4]));
+
+    Label for_body;
+
+    Reg64 reg_indexes = reg_outer_amount;
+    Reg64 reg_const_params = reg_inner_amount;
+    auto init_ptrs_with_offsets = [&](Reg64 pointer, const int64_t *offsets, Reg64 reg_tmp) {
+        for (int j = 0; j < harness_num_dims; j++) {
+            if (jcp.output_dims[j] != 1 && offsets[j] != 0) {
+                h->mov(reg_tmp, offsets[j]);
+                h->imul(reg_tmp, h->ptr[reg_indexes + j * sizeof(size_t)]);
+                h->add(pointer, reg_tmp);
+            }
+        }
+    };
+    std::vector<Reg64> data_ptr_regs(num_params);
+    for (auto i = 0; i < num_params; i++) {
+        data_ptr_regs[i] = Reg64(static_cast<int>(data_ptr_reg_idxs[i]));
+        if (i < num_inputs)
+            h->mov(data_ptr_regs[i], h->ptr[reg_const_params + GET_OFF(src_ptrs) + i * sizeof(void*)]);
+        else
+            h->mov(data_ptr_regs[i], h->ptr[reg_const_params + GET_OFF(dst_ptrs) + (i - num_inputs) * sizeof(void*)]);
+        // we can use the last data_ptr_reg as tmp_reg until the last iteration, and reg_const_params then
+        Reg64 reg_tmp = i < num_params-1 ? Reg64(static_cast<int>(data_ptr_reg_idxs.back())) : reg_const_params;
+        init_ptrs_with_offsets(data_ptr_regs[i], &jcp.data_offsets[i * harness_num_dims], reg_tmp);
+    }
+    auto emit_tiles = [&]() {
+        bool inner_work_amount_is_set = false;
+        auto process_tile =
+                [&](bool body_condition, const std::vector<EmitterCode>& body,
+                                    bool tile_condition, const EmitterCode& tile) {
+            if (body_condition) {
+                // emit Tile body directly if only one tile iteration is needed
+                for (auto& code : body)
+                    code.first->emit_code(code.second.first, code.second.second, vec_pool, gpr_pool);
+            } else if (tile_condition) {
+                // Need to set proper work amount for inner tiles before code emission
+                if (!inner_work_amount_is_set) {
+                    h->mov(reg_inner_amount, inner_work_amount);
+                    inner_work_amount_is_set = true;
+                }
+                std::vector<size_t> in_regs, out_regs;
+                std::tie(in_regs, out_regs) = tile.second;
+                // pass work_amount reg to Tile
+                in_regs.push_back(static_cast<size_t>(reg_inner_amount.getIdx()));
```

Comment:
May be ionly I have some problems here but could you please add some comments for what this part of code? Why we have if-else branches?